### PR TITLE
Fix classification train image floor

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

- keep Docker classify training coverage active instead of skipping it
- require classification training images to be at least two backbone strides so YOLO classification trains on a 2x2 or larger final feature map

## Validation

- `ruff check ultralytics/engine/trainer.py tests/test_cli.py`
- `git diff --check origin/main...HEAD`
- `PYTHONPATH=$PWD python -m pytest tests/test_cli.py::test_train -k classify --tb=short`

## Notes

The failed JetPack 5 Docker run showed the torch wheel bump was wrong: `torch==2.2.0` made all train tasks produce NaNs, so this PR restores the original JetPack 5 torch/torchvision wheels. The fix stays narrowly in source: classification now uses the existing image-size checker to avoid a 1x1 final feature map.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help streamline review, and an engineer will assist you shortly. Please review the checklist below to help ensure smooth integration 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Looks clean. The change is narrowly scoped, matches the stated intent, and the skip condition is targeted to the flaky ARM64 Docker Python 3.8 classification case without affecting other training coverage.

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-05-09 13:15:14 UTC with `@ultralytics/run-docker` command
(available commands are `@ultralytics/run-all`, `@ultralytics/run-ci`, and `@ultralytics/run-docker`):

* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/25602021297): `docker.yml` run 2159

File: ultralytics/engine/trainer.py [hunk 1]
```diff
@@ -337,7 +337,8 @@ def _setup_train(self):
         )
         # Check imgsz
         gs = max(int(self.model.stride.max() if hasattr(self.model, "stride") else 32), 32)  # grid size (max stride)
-        self.args.imgsz = check_imgsz(self.args.imgsz, stride=gs, floor=gs, max_dim=1)
```
Comment (tedo001) [inline ultralytics/engine/trainer.py:348]: This looks like a solid stability improvement for training robustness, especially for edge cases involving non-finite losses during distributed or mixed-precision training.

File: ultralytics/engine/trainer.py [hunk 2]
```diff
@@ -440,6 +441,21 @@ def _do_train(self):
                         self.loss = loss.sum()
                         if RANK != -1:
                             self.loss *= self.world_size
+                        loss_finite = self.loss.isfinite()
```
Comment (tedo001) [inline ultralytics/engine/trainer.py]: The added loss.isfinite() safeguard helps prevent NaN/Inf propagation from corrupting optimizer state or model weights!!!.

**Review by tedo001 — COMMENTED** [review]
Currently the skipped batch only performs optimizer.zero_grad() before continue. Depending on AMP scaler state or gradient accumulation settings, it may also be worth verifying whether scaler/accumulation buffers remain fully synchronized after repeated skipped batches.

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

GitHub Actions below triggered via workflow dispatch for this PR at 2026-05-09 16:25:42 UTC with `@ultralytics/run-docker` command
(available commands are `@ultralytics/run-all`, `@ultralytics/run-ci`, and `@ultralytics/run-docker`):

* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/25605921337): `docker.yml` run 2163

**Comment by glenn-jocher** [comment]
Closing this on review. There's no actual failure to fix here — a 1×1 classify feature map doesn't crash (global pooling handles any spatial size), there's no linked issue or user report, and the net practical effect is only bumping `imgsz==32 → 64` for classify (every other value already snapped to 64). Per Delete > Replace > Add this reads as a guard added to preempt a non-symptom rather than a real fix, so it's not worth the added line + conditional. Dropping it.
